File: add_col.py
import sqlite3
import os

for root, dirs, files in os.walk(r'C:\Negar'):
    for f in files:
        if f.endswith('.db'):
            path = os.path.join(root, f)
            try:
                conn = sqlite3.connect(path)
                cur = conn.cursor()
                cur.execute("PRAGMA table_info(Companies);")
                cols = [c[1] for c in cur.fetchall()]
                if 'CodingType' not in cols:
                    cur.execute("ALTER TABLE Companies ADD COLUMN CodingType TEXT;")
                    conn.commit()
                    print('Added CodingType column to:', path)
                else:
                    print('CodingType already exists in:', path)
                conn.close()
            except Exception as e:
                print('Error in', path, ':', e)

File: AddCol.vb
Imports System
Imports System.Data.SQLite

Module Program
    Sub Main()
        Try
            Dim connStr = "Data Source=C:\Negar\Database\Negar.db;Version=3;"
            Using conn As New SQLiteConnection(connStr)
                conn.Open()
                Using cmd As New SQLiteCommand("ALTER TABLE Companies ADD COLUMN Level6Length INTEGER DEFAULT 2", conn)
                    cmd.ExecuteNonQuery()
                End Using
                Console.WriteLine("Column Level6Length added successfully.")
            End Using
        Catch ex As Exception
            Console.WriteLine("Error: " & ex.Message)
        End Try
    End Sub
End Module
